`changelog-newversion` GitHub action failing after release

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

<!--- Thank you for keeping this note for the community --->

See https://github.com/hashicorp/terraform-provider-awscc/actions/runs/10724303070/job/29739913215:

```
Run git config --local user.email changelogbot@hashicorp.com
  git config --local user.email changelogbot@hashicorp.com
  git config --local user.name changelogbot
  git add CHANGELOG.md
  git commit -m "Update CHANGELOG.md after v1.13.0"
  git push
  shell: /bin/bash -e {0}
[main c9b504c4[2](https://github.com/hashicorp/terraform-provider-awscc/actions/runs/10724303070/job/29739913215#step:4:2)] Update CHANGELOG.md after v1.13.0
 1 file changed, 2 insertions(+), 1 deletion(-)
remote: error: GH01[3](https://github.com/hashicorp/terraform-provider-awscc/actions/runs/10724303070/job/29739913215#step:4:3): Repository rule violations found for refs/heads/main.        
remote: Review all repository rules at http://github.com/hashicorp/terraform-provider-awscc/rules?ref=refs%2Fheads%2Fmain        
remote: 
remote: - Changes must be made through a pull request.        
remote: 
To https://github.com/hashicorp/terraform-provider-awscc
 ! [remote rejected]     main -> main (push declined due to repository rule violations)
error: failed to push some refs to 'https://github.com/hashicorp/terraform-provider-awscc'
Error: Process completed with exit code 1.
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -66,10 +66,16 @@ jobs:
     if: github.ref_name == needs.highest-version-tag.outputs.tag
     runs-on: macos-latest
     steps:
+      - uses: actions/create-github-app-token@3378cda945da322a8db4b193e19d46352ebe2de5 # v1.10.4
+        id: app-token
+        with:
+          app-id: ${{ secrets.APP_ID }}
+          private-key: ${{ secrets.APP_PEM }}
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
           fetch-depth: 0
           ref: main
+          token: ${{ steps.app-token.outputs.token }}
       - name: Update Changelog Header
         run: |
           CHANGELOG_FILE_NAME="CHANGELOG.md"

